### PR TITLE
[Feature] Initial Farming page

### PR DESCRIPTION
**Pull request by Zoma-Ancestral**
The goal is to provide a basic display of W6 Farming skill data, including :
- Bonuses from the market, with level and cost
- Active crops with evo chances, OG chances, and time till the end of the cycle
- All crops with owned quantities
- Alert if some crops are ready to be collected
- Display how many magic beans can be acquired for trading current inventory
- Display Crop Scientist bonuses

Also, mark the beginning of the un-hide w6 stuff era by removing filters to hide some locked w6 data

W6 Skills have been added to Rift

Jade Upgrade that remove the bonus reduction from stamps when under-level for some skilling stamps is now takent into account when retrieving stamps bonus for calculation

### CODE REVIEW

File: data/domain/stamps.tsx [hunk 1]
```diff
@@ -309,6 +313,7 @@ export function updateStamps(data: Map<string, any>) {
         })
     }
 
+    const limitIsRemoved = sneaking.jadeUpgrades.find(upgrade => upgrade.index == 0)?.purchased ?? false;
```
Comment (Sludging) [inline data/domain/stamps.tsx:316]: I think the correct index is 5, unless I'm missing something 🤔 
```suggestion
    const limitIsRemoved = sneaking.jadeUpgrades.find(upgrade => upgrade.index == 5)?.purchased ?? false;
```

File: data/domain/world-6/summoning.tsx [hunk 2]
```diff
@@ -345,7 +354,7 @@ export class Summoning extends Domain {
                 colorBattles = summoning.summonBattles.allBattles[index];
             }
 
-            summoning.summonEssences.push({ color: index, quantity: value, display: shouldDisplay, victories: colorVictories, battles: colorBattles });
+            summoning.summonEssences.push({ color: index, quantity: value, unlocked: unlocked, display: display, victories: colorVictories, battles: colorBattles });
```
Comment (Sludging) [inline data/domain/world-6/summoning.tsx:357]: Why do we need both `unlocked` and `display`? I'm a bit unclear from just looking at this code

File: data/domain/rift.tsx [hunk 3]
```diff
@@ -58,11 +61,10 @@ export class InfiniteStarsBonus extends RiftBonus {
...
 
 // Use a typeguard to prove a skill index is a valid rift mastery skill
-const isRiftMasterySkill = (x: SkillsIndex): x is RiftMasterySkills => 
- ![SkillsIndex.Farming, SkillsIndex.Summoning, SkillsIndex.Sneaking].includes(x)
+const isRiftMasterySkill = (x: SkillsIndex): x is RiftMasterySkills => true;
 
-// Rift doesn't care about the new w6 skills for now, so exclude them from our index.
-type RiftMasterySkills = Exclude<SkillsIndex, SkillsIndex.Farming | SkillsIndex.Summoning | SkillsIndex.Sneaking>
+// Leaving this in case in need of hiding some other skills
+type RiftMasterySkills = SkillsIndex
```
Comment (Sludging) [inline data/domain/rift.tsx:67]: oh I didn't even know they were added to the rift lol.

**Review by Sludging — COMMENTED** [review]
few small comments, nothing big.

I only glanced at farming, it's .. a lot lol. Great work pushing through it!

File: data/domain/world-6/summoning.tsx (hunk 2 above)
Comment (Zoma-Ancestral) [reply data/domain/world-6/summoning.tsx:357]: Display is used to hide last unused yet essences
Unlocked is used to know which essence should be showed in the currency list (avoid showing all and having the locked one at 0 at all times

File: data/domain/rift.tsx (hunk 3 above)
Comment (Zoma-Ancestral) [reply data/domain/rift.tsx:67]: I noticed that on Toolbox when I glanced at it for Farming, then checked in-game and saw that they're indeed in the game x)

**Review by Sludging — APPROVED** [review]
Great work on this! Feel free to merge it at any time.

Really outstanding amount of work lol!

**Comment by Marciland** [comment]
Fyi Looks like this broke something. My idleon efficiency loads forever and becomes unresponsive

network doesnt show any outgoing requests after document is 200.

**Pull request merged**